bc_obps/reporting/models/triggers.py

```diff
@@ -0,0 +1,75 @@
+from typing import Iterator
+import pgtrigger
+
+
+def get_sql_relation_alias_generator() -> Iterator[str]:
+    i = 0
+    while True:
+        i += 1
+        yield f"rel{str(i)}"
+
+
+def immutable_report_version_trigger(
```

Comment:
This generates SQL to apply as a trigger.
The entire expected SQL can be found in the test file for this method